dmoj/checkers/bridged.py

```diff
@@ -0,0 +1,122 @@
+from dmoj.executors import executors
+from dmoj.executors.base_executor import CompiledExecutor
+from dmoj.judgeenv import env, get_problem_root
+from dmoj.result import CheckerResult
+from dmoj.utils import ansi
+from dmoj.utils.unicode import utf8text
+
+checker_defaults = {
+    'time_limit': env['generator_time_limit'],
+    'memory_limit': env['generator_memory_limit'],
+    'compiler_time_limit': env['compiler_time_limit'],
+    'feedback': True
+}
+
+executor = None
+
+def get_executor(checker_kwargs, problem_id):
+    global executor
+
+    if executor is None:
+        if 'files' not in checker_kwargs:
+            raise InternalError('No checker file[s] specified!')
+        if 'lang' not in checker_kwargs:
+            raise InternalError('Language not specified for checker!')
+
+        filenames = list(map(lambda x: os.path.join(get_problem_root(problem_id), x), checker_kwargs['files']))
+        for filename in filenames:
+            print(os.path.abspath(filename))
+        sources = {}
+
+        try:
+            for filename in filenames:
+                with open(filename, 'r') as f:
+                    sources[os.path.basename(filename)] = f.read()
+        except:
+            traceback.print_exc()
+            raise IOError('Could not read checker source!')
+
+        use_c_or_cpp = any(map(lambda name: os.path.splitext(name)[1] in ['.c', '.cpp'], filenames))
+
+        clazz = executors.get(checker_kwargs['lang'])
+        if clazz is None:
+            raise InternalError('Could not find an executor for language "%s"' % checker_kwargs['lang'])
+
+        clazz = clazz.Executor
+
+        fs = clazz.fs + [tempfile.gettempdir()]
+        clazz = type('Executor', (clazz,), {'fs': fs})
+
+        if issubclass(clazz, CompiledExecutor):
+            compiler_time_limit = checker_kwargs['compiler_time_limit'] or clazz.compiler_time_limit
+            clazz = type('Executor', (clazz,), {'compiler_time_limit': compiler_time_limit})
+
+        if hasattr(clazz, 'flags'):
+            flags = checker_kwargs.get('flags', [])
+
+            flags += ['-DWINDOWS_JUDGE', '-DWIN32'] if os.name == 'nt' else ['-DLINUX_JUDGE']
+
+            # We shouldn't be mutating the base class flags.
+            # See <https://github.com/DMOJ/judge/issues/174>.
+            clazz = type('FlaggedExecutor', (clazz,), {'flags': flags + list(clazz.flags)})
+
+        try:
+            # Optimize the common case.
+            if use_c_or_cpp:
+                # Some checkers (like those using testlib.h) take an extremely long time
+                # to compile, so we cache them.
+                executor = clazz('_checker', None, aux_sources=sources, cached=True)
+            else:
+                if len(sources) > 1:
+                    raise InternalError('non-C/C++ checker cannot be multi-file')
+                executor = clazz('_checker', list(sources.values())[0])
+        except CompileError as err:
```

Comment:
This branch is no longer necessary after #489, you can just let the exception bubble up.